andela/kenya-cfh
# feature(implement token on signup): provide user with token on signup

#### What does this PR do?
Enable the provision of token to users on signup
#### Description of Task to be completed?
- install lodash, chai and jsonwebtoken dependencies
- write custom validations for validating user input
- write unit test for signup features
- modify test script in package.json
- write controllers for client-side and server-side
- modify game.js file
#### How should this be manually tested?
After cloning to local machine, change directory into kenya-cfh and run `npm install` to install dependencies.
- run `npm run heroku-postbuild` to build the application
- run `npm test` to ensure all tests are passing
- run `npm start` to startup application and open browser at localhost:3000. Click on sign-up and fill in the user details
- To test with Postman, enter the url `localhost:3000/api/auth/signup` and enter the parameters `name`, `email` and `password`.
#### Any background context you want to provide?
- N/A
#### What are the relevant pivotal tracker stories?
#153765403
#### Screenshots (if appropriate)
<img width="1440" alt="token" src="https://user-images.githubusercontent.com/24798364/34787596-1395e6bc-f638-11e7-8325-774b768febc8.png">



<img width="1440" alt="screen shot 2018-01-10 at 2 32 45 pm" src="https://user-images.githubusercontent.com/24798364/34787624-313f1238-f638-11e7-9f18-dcaa3d5c3f81.png">



<img width="1440" alt="error2" src="https://user-images.githubusercontent.com/24798364/34787712-89217fb8-f638-11e7-9d2c-bf4b11ed12fe.png">



<img width="1440" alt="error1" src="https://user-images.githubusercontent.com/24798364/34787738-a1659cb2-f638-11e7-8f53-d1a5d7b9efa7.png">






#### Questions:
- N/A
